src/worker/jobs/downloads/update_metadata.rs

```diff
@@ -54,6 +83,83 @@ fn update(conn: &mut PgConnection) -> QueryResult<()> {
     Ok(())
 }
 
+fn batch_update(batch_size: i64, conn: &mut PgConnection) -> QueryResult<i64> {
+    table! {
+        /// Imaginary table to make Diesel happy when using the `sql_query` function.
+        sql_query_results (count) {
+            count -> BigInt,
+        }
+    }
+
+    /// A helper struct for the result of the query.
+    ///
+    /// The result of `sql_query` can not be a tuple, so we have to define a
+    /// proper struct for the result.
+    #[derive(QueryableByName)]
+    struct SqlQueryResult {
+        count: i64,
+    }
+
+    let query = format!(
+        r#"
+            WITH downloads_batch AS (
+                -- Select a batch of downloads (incl. `crate_id`) that have not been
+                -- counted yet.
+                SELECT crate_id, version_id, date, version_downloads.downloads - counted AS downloads
+                FROM version_downloads
+                INNER JOIN versions ON versions.id = version_id
+                WHERE NOT processed AND version_downloads.downloads != counted
+                LIMIT {batch_size}
```

Comment:
_eye twitches very slightly at string interpolation in a query, even though it's obviously safe here_

Reply:
I tried binding it, but for whatever reason it resulted in an SQL syntax error that I couldn't figure out.

Reply:
You could change the placeholder from `?` to `$1` then it should be fine.

Reply:
indeed, that worked. thanks! :)